### PR TITLE
test(ModalHeader): migrate to React testing library

### PR DESCRIPTION
**Pull request by minimatrix**
- [ ] Bug fix 
- [ ] New feature 
- [x] Chore 
- [ ] Breaking change 
- [x] There is an open issue which this change addresses
- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** document.
- [x] My commits follow the [Git Commit Guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] My change requires a change to [Typescript typings](./types/lib).
  - [ ] I have updated the typings accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

#2625

### CODE REVIEW

**Comment by minimatrix** [comment]
I made this PR as part of hacktoberfest, if you are happy with the work I have done please could you add the hacktoberfest-accepted label. Thanks for the opportunity to contribute!

**Comment by illiteratewriter** [comment]
@minimatrix I've requested some changes to be made. 

It would be great if you can fix them and submit them so that you're eligible for hactoberfest.

**Comment by minimatrix** [comment]
> @minimatrix I've requested some changes to be made.
> 
> It would be great if you can fix them and submit them so that you're eligible for hactoberfest.

I'm afraid I can't see the changes that you've requested

File: src/__tests__/ModalHeader.spec.js [hunk 1]
```diff
@@ -1,44 +1,55 @@
...
+        </ModalHeader>
+      );
+      expect(screen.getByTestId('modalheader-id')).toHaveClass('modal-header');
+      expect(screen.getByTestId('modalheader-id')).toHaveClass('other');
   });
 
   it('should render close button', () => {
-    const wrapper = shallow(
-      <ModalHeader toggle={() => {}} className="other">
+    const {container} = render(
```
Comment (davidacevedo) [inline src/__tests__/ModalHeader.spec.js:29]: You should just be using `screen` vs container here

File: src/__tests__/ModalHeader.spec.js [hunk 2]
```diff
@@ -1,44 +1,55 @@
...
+      <ModalHeader toggle={() => {}} className="other" data-testid="modalheader-id">
         Yo!
-      </ModalHeader>,
+      </ModalHeader>
     );
-
-    expect(wrapper.hasClass('other')).toBe(true);
-    expect(wrapper.hasClass('modal-header')).toBe(true);
-    expect(wrapper.find('button.btn-close').length).toBe(1);
+    expect(container.getElementsByClassName('btn-close')).toHaveLength(1);
```
Comment (davidacevedo) [inline src/__tests__/ModalHeader.spec.js:34]: You can use `container.getByRole('button')` here instead

File: src/__tests__/ModalHeader.spec.js [hunk 3]
```diff
@@ -1,44 +1,55 @@
...
+    expect(screen.getByTestId('modalheader-id')).toHaveClass('modal-header');
+    expect(screen.getByTestId('modalheader-id')).toHaveClass('other');
   });
 
   it('should render custom tag', () => {
-    const wrapper = shallow(<ModalHeader tag="p">Yo!</ModalHeader>).childAt(0);
+    render(<ModalHeader tag="p" data-testid="modalheader-id">Yo!</ModalHeader>);
+    expect(screen.getByTestId('modalheader-id')).toHaveTextContent('Yo!');
+    expect(screen.getByTestId('modalheader-id').firstChild.tagName.toLowerCase()).toMatch('p');
```
Comment (davidacevedo) [inline src/__tests__/ModalHeader.spec.js:42]: You can use `document.querySelector` for the `p` tag here.

File: src/__tests__/ModalHeader.spec.js [hunk 4]
```diff
@@ -1,44 +1,55 @@
...
   it('should render custom wrapping tag', () => {
-    const wrapper = shallow(<ModalHeader wrapTag="main">Yo!</ModalHeader>);
-
-    expect(wrapper.type()).toBe('main');
+    render(
+      <ModalHeader wrapTag="main" data-testid="modalheader-id">
+        Yo!
+      </ModalHeader>
+    );
+    expect(screen.getByTestId('modalheader-id').tagName.toLowerCase()).toMatch('main');
```
Comment (davidacevedo) [inline src/__tests__/ModalHeader.spec.js:52]: Same as above `document.querySelector('main')`

**Comment by davidacevedo** [comment]
> I made this PR as part of hacktoberfest, if you are happy with the work I have done please could you add the hacktoberfest-accepted label. Thanks for the opportunity to contribute!

Sorry for missing the deadline for the hacktoberfest! I can take up the work for this if you are no longer interested in keeping it going

**Pull request closed without merging**